bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/util/Converter.java

```diff
@@ -39,8 +39,13 @@ public class Converter {
         if (dateTimeString == null) {
             return null;
         }
-        String fixedTs = dateTimeString.substring(0, 1).toUpperCase() + dateTimeString.substring(1);
-        return LocalDateTime.parse(fixedTs, ROUTEROS_FORMAT);
+        // As of Firmware 7.10 the date format has changed to "yyyy-MM-dd HH:mm:SS"
```

Comment:
Perhaps a [DateTimeFormatter](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/format/DateTimeFormatter.html) could also be used for this new format like it's used for the old one? Something like:

```java
private static final DateTimeFormatter ROUTEROS_NEW_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss", Locale.ENGLISH);

// As of Firmware 7.10 the date format has changed to "yyyy-MM-dd HH:mm:ss"
try {
    return LocalDateTime.parse(fixedTs, ROUTEROS_NEW_FORMAT);
} catch (DateTimeParseException) {
    // Fallback to old format.
}

try {
    String fixedTs = dateTimeString.substring(0, 1).toUpperCase() + dateTimeString.substring(1);
    return LocalDateTime.parse(fixedTs, ROUTEROS_FORMAT);
} catch (DateTimeParseException) {
    return null;
}
```

I don't know if the caller will handle `DateTimeParseException` gracefully, that would probably be preferable over returning `null` for detailed logging.

It seems the existing code returns `@Nullable` but also doesn't handle `DateTimeParseException` internally, which looks a bit inconsistent.

Reply:
Indeed the existing code is not handling a `DateTimeParseException`, so i changed it to return `null`, thanks!
For the fallback i'd prefer to check for string length instead of another try/catch block. I don't see what another dateformattter would add, as the new format is a valid ISO-8601 sting that is parsed by default with `LocalDateTime.parse`. But i might be missing something here.

Added another check for strings that are too short anyway and added some tests, just ot be sure.

Reply:
> For the fallback i'd prefer to check for string length instead of another try/catch block. I don't see what another dateformattter would add, as the new format is a valid ISO-8601 sting that is parsed by default with `LocalDateTime.parse`. But i might be missing something here.

The only reason would be that internally [LocalDateTime.parse(CharSequence text)](https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#parse-java.lang.CharSequence-) already uses [DateTimeFormatter.ISO_LOCAL_DATE_TIME](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_LOCAL_DATE_TIME). 

Therefore it would be faster to directly use a tailored `DateTimeFormatter` (set up once) rather than performing custom string operations before using another one that doesn't fully fit the format.

This was only a minor proposal for consideration, so I will definitely not insist. 🙂

Reply:
Thanks for the clarification, changed it.

Reply:
If you are curious, you could also try to run the fallback version without the string length check. 🙂 It will probably be much slower if most strings are in the old format (because exception handling is slow), but will be faster when on the new firmware. Btw, I think the build failed because of the empty string test case.